### PR TITLE
Enable inviting second admin during setup flow

### PR DESCRIPTION
**Pull request by noahmoss**
Adds the ability to send `invite` details in the `POST /api/setup` request that is sent at the end of the setup flow. If this is provided, and email is configured (i.e. on cloud instances), we will create a second admin user and send an invite email to them. The copy in the email is modified for this case, based on the value of the `sentFromSetup` parameter in the mustache template.

Also adds a `has_invited_second_user` field to the response to `GET /api/user/current`. This is true iff the current user has the ID of 1 (i.e. they were the user who went through the setup flow) and at least one other user exists (i.e. they have invited another user). This is just used to [tweak the wording in the homepage banner](https://www.notion.so/metabase/Enable-users-to-invite-teammates-with-db-credentials-in-set-up-2e484208a59d4c11815c1e81b98b92c0#7d682d9673584d6a973cebcf7dd6e36c) for this user ("invite a user" -> "invite another user").

Can be tested end-to-end by merging in the frontend changes from https://github.com/metabase/metabase/pull/19404 and following the setup instructions from that PR.

Epic #19247

### CODE REVIEW

**Review by ranquild — APPROVED** [review]
Works correctly with the frontend changes

File: src/metabase/api/setup.clj [hunk 1]
```diff
@@ -56,6 +58,13 @@
       ;; return user ID, session ID, and the Session object itself
       {:session-id session-id, :user-id user-id, :session session})))
 
+(defn- setup-maybe-create-and-invite-user! [{:keys [email first_name last_name] :as user}, invitor]
+  (when email
+    (if-not (email/email-configured?)
+      (log/error (trs "Could not invite user because email is not configured."))
+      (u/prog1 (user/create-and-invite-user! user invitor true)
+        (user/set-permissions-groups! <> [(group/all-users) (group/admin)])))))
```
Comment (dpsutton) [inline src/metabase/api/setup.clj:66]: this line seems really heavy: `set-permissions-groups all-users admin`. It seems we have user and invitor, can we use that to set those permissions? I know we have some predicates that assume the entire userbase is user and inviter, but if this function gets reused somewhere it has enormous implications.

Comment (noahmoss) [reply src/metabase/api/setup.clj:66]: > It seems we have user and invitor, can we use that to set those permissions?

Could you clarify what you mean by this?

I'm following similar logic here as how it's done in the `POST /api/user` endpoint for creating new users normally. (First a user is created, then permission groups are added).

Would it help if I renamed this `setup-maybe-create-and-invite-admin!` so it's clear that the function adds the user to the admin group?

Comment (dpsutton) [reply src/metabase/api/setup.clj:66]: Ah i was misreading that line. I forgot `group/all-users` is a group. I was thinking this was setting the permissions of every user on the instance to be an admin with the assumption that there were only two users.

File: src/metabase/models/user.clj [hunk 2]
```diff
@@ -152,17 +152,28 @@
...
+(defn add-has-invited-second-user
+  "Adds the `has_invited_second_user` flag to a collection of `users`. This should be `true` for only the user who
+  underwent the initial app setup flow (with an ID of 1), iff more than one user exists. This is used to modify
+  the wording for this user on a homepage banner that prompts them to add their database."
+  {:batched-hydrate :has_invited_second_user}
+  [users]
+  (when (seq users)
+    (for [user users]
+      (assoc user :has_invited_second_user (and (= (:id user) 1)
+                                                (> (db/count User) 1))))))
```
Comment (dpsutton) [inline src/metabase/models/user.clj:164]: this `(db/count User)` should not be in the for experssion. We can get this count once and then use it.

Comment (noahmoss) [reply src/metabase/models/user.clj:164]: good call, will fix

**Review by dpsutton — APPROVED** [review]
remove the potential for N+1 `db/count User` and we're good to go.

**Pull request merged**